### PR TITLE
globals convert to responsive bootstrap, step 2

### PR DESCRIPTION
**Pull request by bradymiller**
Converted globals to bootstrap. It is responsive on the main globals gui. The user settings stays in table format.

TODO:
1. Search is a bit funky. It finds the items well. But the highlighting of the tabs is not correct.
2. More testing

### CODE REVIEW

**Comment by mdsupport** [comment]
It would be good to get rid of `tabbify` and convert current menu bar to `nav`.  That conversion will also highlight the fact that menu has become too granular from ui design perspective.  Something you could experiment would be a nav bar as -

`title` `prev arrow` `Setting Group: ` `group dropdown` `next arrow` [`save button`] `search box+button`

**Comment by mdsupport** [comment]
On a related note, strongly recommend making available [boothelp](https://github.com/cobisja/boothelp)  or any other bootstrap code injector classes as an design component vs developing something from scratch.

**Comment by robertdown** [comment]
@bradymiller Let's work together on this PR. I have a couple other things I want to do to the globals page

**Comment by bradymiller** [comment]
@mdsupport and @robertdown ,
Sounds good. All the heavy lifting is done and I just have a bug or 2 to work out, so feel free to work on this PR. I am guessing that Robert has a sidebar menu in the works which would also address mdsupport's suggestions.

**Comment by robertdown** [comment]
Integrating a sidebar menu for sure, actual navbar as well. @mdsupport We'll look into Boothelp. Also, I recently introduced event dispatching. This will allow modules to better manage how they to interact with the admin screen.

**Comment by bradymiller** [comment]
also word of warning on this script :)
This is a rather complicated script that functions as the Administration->Globals and the User->Settings along with some other items that need support (search mechanism, portal package download, etc). This is why I took the least invasive approach I could.

**Comment by mdsupport** [comment]
Try the following highlighter:
```
  <?php // mdsupport - Highlight search results ?>
  $('.srch div').wrapInner("<mark></mark>");
  $('.tab .row.srch :first-child').closest('.tab').each(function() {
      $('.tabNav li:nth-child('+($(this).index()+1)+') a').wrapInner("<mark></mark>");
  });
```

**Comment by mdsupport** [comment]
BTW, by its nature this screen is to be used by 'admin' types.  What are the cons of doing a ajax update of every change - no update button, may be a check mark appears and fades away.

**Comment by bradymiller** [comment]
@mdsupport ,
I think there are some changes in there that may possibly shut the user out. Would need to analyze that a bit.

File: interface/super/edit_globals.php [hunk 1]
```diff
@@ -383,399 +367,441 @@ function validate_file(){
...
 <?php } else { ?>
-  <form method='post' name='theform' id='theform' action='edit_globals.php' onsubmit='return top.restoreSession()'>
+    <div class="body_top">
 <?php } ?>
-
+<div class="navbar navbar-default navbar-fixed-top">
+    <div class="container-fluid">
+        <div class="navbar-header">
+            <button type="button" class="navbar-toggle visible-xs" data-toggle="sidebar" data-target=".sidebar" style="float: left;">
+                <span class="sr-only">Toggle navigation</span>
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: translate

File: interface/super/edit_globals.php [hunk 2]
```diff
@@ -383,399 +367,441 @@ function validate_file(){
...
+                <i class="fa fa-bars"></i>
+            </button>
+            <button type="button" class="navbar-toggle visible-xs" data-toggle="sidebar" data-target="main-navbar">
+                <span class="sr-only">{{ "Toggle Navigation" }}</span>
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: translate

**Comment by bradymiller** [comment]
Search and Save buttons on top are not working.

**Comment by bradymiller** [comment]
@mdsupport , Thanks for the highlighter code fix. With Robert's changes, this will possibly change again, but it was very helpful for me to figure out how to travel in this DOM via jquery.

File: interface/super/edit_globals.php [hunk 3]
```diff
@@ -376,11 +376,11 @@ function validate_file(){
     <div class="container-fluid">
         <div class="navbar-header">
             <button type="button" class="navbar-toggle visible-xs" data-toggle="sidebar" data-target=".sidebar" style="float: left;">
-                <span class="sr-only">Toggle navigation</span>
+                <span class="sr-only"><?php echo xl("Toggle Navigation");?></span>
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: xlt

File: interface/super/edit_globals.php [hunk 4]
```diff
@@ -376,11 +376,11 @@ function validate_file(){
...
                 <i class="fa fa-bars"></i>
             </button>
             <button type="button" class="navbar-toggle visible-xs" data-toggle="sidebar" data-target="main-navbar">
-                <span class="sr-only">{{ "Toggle Navigation" }}</span>
+                <span class="sr-only"><?php echo xl("Toggle Navigation");?>s</span>
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: xlt

**Comment by robertdown** [comment]
FYI, my latest push has dummy code I have to pull. But if you want to see the event dispatcher hard at work (albeit in a very rudamentry way) check out the CalendarSubscriber.php file and then tinker around the actual globals UI - you should notice a menu item called "AAA" at the top

File: library/admin/Resources/config/adminMenu.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * This file is part of OpenEMR.
+ *
+ * @href https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+return [
```
Comment (bradymiller) [inline library/admin/Resources/config/adminMenu.php]: isn't this redundant to what is already in library/globals.php ?

**Comment by bradymiller** [comment]
@robertdown ,
Just submitted fix for the user settings save. The main error was that the toggle settings throws a null error which goes away when wrap it with trim. I am hoping the globals save will be the same simple fix.
-brady

**Comment by bradymiller** [comment]
@robertdown ,
Just checked in a fix for the globals bug :)
-brady

File: library/admin/Resources/config/adminMenu.php (hunk 5 above)
Comment (bradymiller) [reply library/admin/Resources/config/adminMenu.php]: @robertdown ,
I am still not clear why we have this redundant list?

File: interface/super/edit_globals.php [hunk 6]
```diff
@@ -2,179 +2,67 @@
...
-            $couch->createView($GLOBALS['couchdb_dbase']);
-        }
-    }
+$globalsService = new Globals($GLOBALS, new Filesystem());
 
-    return true;
-}
+global $GLOBALS_METADATA;
+global $USER_SPECIFIC_TABS;
+global $USER_SPECIFIC_GLOBALS;
```
Comment (bradymiller) [inline interface/super/edit_globals.php]: Are these globals needed anymore? It looks like  you bring them into your functions as parameters.

File: library/admin/src/Service/Globals.php [hunk 7]
```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * This file is part of OpenEMR.
+ *
+ * @link https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
```
Comment (bradymiller) [inline library/admin/src/Service/Globals.php]: this needs a full license/copyright/author header

Comment (bradymiller) [reply library/admin/src/Service/Globals.php]: Also, shouldn't we be avoiding caps in file/dir names (except for class files of course)?

Comment (bradymiller) [reply library/admin/src/Service/Globals.php]: @robertdown ,
Also, see my comments directly above (i just realized this is my PR and you don't get these comments automatically)

File: composer.json [hunk 8]
```diff
@@ -30,6 +30,7 @@
         "symfony/config": "2.8.*",
         "symfony/dependency-injection": "2.8.*",
         "symfony/event-dispatcher": "2.8.*",
+        "symfony/filesystem": "2.8.*",
```
Comment (bradymiller) [inline composer.json]: @robertdown ,
To avoid merge conflict pain, recommend just bringing this composer package into main codebase now.

**Comment by bradymiller** [comment]
Broke this PR into 2 steps. 1st step is the initial bootstrap conversion and second is the code modernization. Will bring in step 1 at this PR (thank you @mdsupport for the searching solution!):
https://github.com/openemr/openemr/pull/1334
And will continue to work on step 2 in this PR.

**Pull request closed without merging**